docs/parallel_execution.adoc

```diff
@@ -0,0 +1,744 @@
+  }
+}
+</style>
+skinparam shadowing false
+* SpockEngine
+** @ResourceLock("S")\nSpecA <<write_lock>>
+*** @ResourceLock("A")\nTestA1 <<write_lock>>
+*** @ResourceLock("B")\nTestA2 <<write_lock>>
+@endwbs
+....
+
+
+.Lock coarsening - after
+[[figure-lock-coarsening-after]]
+[plantuml,images/lock-coarsening-after,svg]
+....
+@startwbs
+<style>
+wbsDiagram {
+  BackgroundColor #88CCEE
+  RoundCorner 10
+  node {
+    :depth(0) {
+      BackgroundColor #DDCC77
+      RoundCorner 0
+    }
+  }
+
+  .samethread * {
+      BackgroundColor #DDCC77
+      RoundCorner 0
+  }
+
+  .write_lock * {
+      LineStyle 4
+      LineThickness 2.5
+      LineColor #117733
+  }
+}
+</style>
+skinparam shadowing false
+* SpockEngine
+** @ResourceLock("A")\n@ResourceLock("B")\n@ResourceLock("S")\nSpecA <<write_lock>>
+*** TestA1 <<samethread>>
+*** TestA2 <<samethread>>
+@endwbs
+....
+
+=== Isolated Execution
+
+Sometimes, you want to modify and test something that affects every other feature, you could put a `READ` `@ResourceLock` on _every_ feature, but that is impractical.
+The `@Isolated` extension enforces, that only this feature runs without any other features running at the same time.
+You can think of this as an implicit global lock.
+
+As with other locks, the features in an `@Isolated` specification will run in `SAME_THREAD` mode.
+`@Isolated` can only be applied at the specification level so if you have a large specification and only need it for a few features,
+you might want to consider splitting the spec into `@Isolated` and non isolated.
+
+
+.`@Isolated` execution
+[[figure-isolated]]
+[plantuml,images/isolated,svg]
+....
+@startwbs
+<style>
+wbsDiagram {
+  BackgroundColor #88CCEE
+  RoundCorner 10
+  node {
+    :depth(0) {
+      BackgroundColor #DDCC77
+      RoundCorner 0
+    }
+  }
+
+  .isolated * {
+      BackgroundColor #CC6677
+      RoundCorner 0
+      LineStyle 8
+      LineThickness 2.5
+      LineColor #117733
+  }
+}
+</style>
+skinparam shadowing false
+* SpockEngine
+** @Isolated\nSpecA  <<isolated>>
+*** TestA1
+*** TestA2
+** SpecB
+*** TestB1
+*** TestB2
+**** TestB2[1]
+**** TestB2[2]
+@endwbs
+....
+
+[[parallel-thread-pool]]
+== Parallel Thread Pool
+
+With parallel execution enabled, specifications and features can execute concurrently.
+You can control the size of the thread pool that executes the features.
+Spock uses `Runtime.getRuntime().availableProcessors()` to determine the available processors.
+
+* `dynamic(BigDecimal factor)` - Computes the desired parallelism based on the number of available processors multiplied by the `factor` and rounded down to the nearest integer.
+  For example a factor of `0.5` will use half your processors.
+* `dynamicWithReservedProcessors(BigDecimal factor, int reservedProcessors)` - Same as `dynamic` but ensures that the given amount of `reservedProcessors` is not used.
+  The `reservedProcessors` are counted against the available cores not the result of the factor.
```

Comment:
I think it is a useful option, especially if you get system with more and more processors, that is also why Spock uses it by default. 

However, Spock wouldn't use the platform implementation, since that one is controlled via properties and Spocks configuration is done via these config scripts.